operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -221,11 +221,29 @@ type CassandraDatacenterSpec struct {
 	// Cassandra users to bootstrap
 	Users []CassandraUser `json:"users,omitempty"`
 
+	Networking *NetworkingConfig `json:"networking,omitempty"`
+
 	AdditionalSeeds []string `json:"additionalSeeds,omitempty"`
 
 	Reaper *ReaperConfig `json:"reaper,omitempty"`
 }
 
+type NetworkingConfig struct {
+	NodePort *NodePortConfig `json:"nodePort,omitempty"`
+}
+
+type NodePortConfig struct {
+	CqlPort          int `json:"cqlPort,omitempty"`
+	CqlSslPort       int `json:"cqlSslPort,omitempty"`
+	BroadcastPort    int `json:"broadcastPort,omitempty"`
+	BroadcastSslPort int `json:"broadcastSslPort,omitempty"`
```

Comment:
maybe remove the word `Port` from the field names here - a JSON path would look like `networking.NodePort.CqlPort` right now?